gen/gen.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"reflect"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+type Data struct {
+	Name string
+	*Yml
+}
+
+func GenCHeader(data *Data, dst io.Writer) error {
+	t := template.
+		New("").
+		Funcs(template.FuncMap{
+			"SComment":     func(v string, indent int) string { return Comment(v, CommentTypeSingleLine, indent, true) },
```

Comment:
An alternative to maintaining absolute indentations in the template is to have two functions to increment and decrement indentation. I've also used special characters in the template to increment and decrement indentation which is then processed at the end of the template.

Not asking for a change here, just suggesting alternatives for you to consider.